src/main/java/util/StringAddCalculator.java

```diff
@@ -5,49 +5,60 @@
 
 public class StringAddCalculator {
 
-    public static final String COMMA = ",";
+    private static final Pattern PATTERN_DELIMITERS = Pattern.compile("[,:]");
+    private static final Pattern PATTERN_CUSTOM = Pattern.compile("//(.)\n(.*)");
+    private static final int ZERO = 0;
+    private static final String MESSAGE_INVALID_CHARACTER_FORMAT = "Invalid character format.";
+    private static final String MESSAGE_NOT_POSITIVE_INT = "For input only positive integers.";
 
     public static int splitAndSum(String str) {
-        int result = 0;
-        if( str == null || str.isEmpty() ) return result;
-        if( str.length() == 1 ) return checkOneNumber(str);
-        if( str.contains(COMMA) && !str.contains(":")) return checkComma(str);
-        if(str.contains(COMMA) && str.contains(":") ) return checkCommaAndDelimiter(str);
-        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(str);
-        if(m.find()) return checkCustomDelimiter(m);
+        if( str == null || str.isEmpty() ) {
+            return ZERO;
+        }
+        int result = ZERO;
+        if( str.length() == 1 ) {
+            result = checkOneNumber(str);
+        }
+        if( PATTERN_DELIMITERS.matcher(str).find() ) {
+            result = checkSeparatorAndDelimiter(str);
+        }
+        Matcher matcher = PATTERN_CUSTOM.matcher(str);
+        if( matcher.find() ) {
+            result = checkCustomDelimiter(matcher);
+        }
+        if( result == ZERO ) {
+            throw new IllegalArgumentException(MESSAGE_INVALID_CHARACTER_FORMAT);
+        }
         return result;
```

Comment:
```suggestion
        if( str == null || str.isEmpty() ) {
            return ZERO;
        }
        
        if( str.length() == 1 ) {
            result = checkOneNumber(str);
        }
        
        Matcher matcher = PATTERN_CUSTOM.matcher(str);
        if( matcher.find() ) {
            return checkCustomDelimiter(matcher);
        }
        
        return checkSeparatorAndDelimiter(str);
```

제가 말씀드렸던 부분은 이런 순서로 진행되면 될 것 같다는 부분이었습니다! 
`checkSeparatorAndDelimiter` 이게 기본 정책이어서, 
빈문자열 체크, 길이 하나 체크, 커스텀 체크 를 모두 통과하면 기본 정책에 따라 처리되어야 할테니까요 ! 

옳지 않은 형식의 문자열을 입력했다면 
아래의 기본정책 로직을 수행하면서 자연스럽게 예외가 발생할 것 같습니다 :) 
```
private static int checkSeparatorAndDelimiter(String str) {
         String[] token = PATTERN_DELIMITERS.split(str);
         return addStringArray(token);
 }
```